static/app/views/performance/traceDetails/transactionBar.tsx

```diff
@@ -462,22 +509,26 @@ class TransactionBar extends Component<Props, State> {
         }}
       >
         {this.renderPerformanceIssues()}
-        <DurationPill
-          durationDisplay={getDurationDisplay({
-            left: startPercentage,
-            width: widthPercentage,
-          })}
-          showDetail={showDetail}
-        >
-          {getHumanDuration(duration)}
-        </DurationPill>
+        {isTraceError(transaction) ? (
+          <ErrorBadge />
+        ) : (
+          <DurationPill
+            durationDisplay={getDurationDisplay({
+              left: startPercentage,
+              width: widthPercentage,
+            })}
+            showDetail={showDetail}
+          >
+            {getHumanDuration(duration)}
+          </DurationPill>
+        )}
       </RowRectangle>
     );
   }
 
   renderPerformanceIssues() {
     const {transaction, barColor} = this.props;
-    if (!isTraceFullDetailed(transaction)) {
+    if (isTraceError(transaction) || isTraceRoot(transaction)) {
```

Comment:
In a lot of places I'm seeing that we're switching `!isTraceFullDetailed` -> `isTraceError(transaction || isTraceRoot(transaction)` and I'm having a hard time drawing conclusions to _why_ we're doing this. Can you help me understand why this conversion is necessary?